### PR TITLE
Order prop

### PR DESCRIPTION
**Pull request by reed-jones**
SearchParams already had order prop and selectedColumn, but now deep watches the object and updates on change

### CODE REVIEW

File: components/Table.vue [hunk 1]
```diff
@@ -150,16 +136,14 @@
...
-            this.defaultParams.order = this.order ? this.order.toLowerCase() : 'asc';
+            this.defaultParams.selectedCol = this.searchParams.selectedCol ? this.searchParams.selectedCol : null;
             this.defaultParams.paginate = this.paginate;
-
-            if (this.url) {
-                this.update(this.url);
-            } else {
-                this.internalData = this.data;
-                this.filteredData = this.data;
+            if (this.searchParams.order) {
```
Comment (kotarCreative) [inline components/Table.vue]: I think you should just loop through all the keys on the `searchParams` prop. That way you don't have to check keys that don't exist.

File: components/Table.vue [hunk 2]
```diff
@@ -187,8 +171,18 @@
         },
 
         watch: {
-            searchParams() {
-                this.search();
+            searchParams: {
+                handler: function(n) {
+                    if (n.selectedCol !== this.defaultParams.selectedCol) {
```
Comment (kotarCreative) [inline components/Table.vue]: Again you can probably just loop through the keys in the `n` variable to check what actually exists.

File: components/Table.vue [hunk 3]
```diff
@@ -198,10 +192,19 @@
...
-            }
+            },
         },
 
         methods: {
+            refresh() {
+                if (this.url) {
+                    this.update(this.url);
+                } else {
+                    this.internalData = val;
```
Comment (kotarCreative) [inline components/Table.vue]: Where is `val` coming from?

Comment (codymoorhouse) [reply components/Table.vue]: Probably should be 
```
this.internalData = this.data;
this.filteredData = this.data;
```

Comment (reed-jones) [reply components/Table.vue]: Oops, nice catch

File: components/Table.vue [hunk 4]
```diff
@@ -173,13 +173,9 @@
         watch: {
             searchParams: {
                 handler: function(n) {
-                    if (n.selectedCol !== this.defaultParams.selectedCol) {
-                        this.defaultParams.selectedCol = n.selectedCol
-                    }
-                    if (n.order) {
-                        this.defaultParams.order = n.order
+                    for (const key of Object.keys(this.searchParams)) {
```
Comment (kotarCreative) [inline components/Table.vue]: You should probably use `n` rather than `this.searchParams`

File: components/Table.vue [hunk 5]
```diff
@@ -136,10 +136,10 @@
         }),
 
         mounted() {
-            this.defaultParams.selectedCol = this.searchParams.selectedCol ? this.searchParams.selectedCol : null;
             this.defaultParams.paginate = this.paginate;
-            if (this.searchParams.order) {
-                this.defaultParams.order = this.searchParams.order
+
+            for (const key of Object.keys(this.searchParams)) {
```
Comment (kotarCreative) [inline components/Table.vue]: Does the watcher not catch this case?

Comment (reed-jones) [reply components/Table.vue]: No, it doesn't seem to

**Pull request merged**